### ISSUE TITLE
Get parameters in url auto populating

### ISSUE THREAD

**Issue by ysrinu** [susanBuck/dwa15-fall2017#109]
I didn't use '[old](https://github.com/susanBuck/dwa15-fall2017/blob/master/03_Laravel/21_Validation.md#retaining-form-data-when-validation-fails)'  in my form.
My form action is GET.
After a form is submitted I can see the name-value pairs in the url (following ?).
Now, when I clear the url and only type the base url, I would expect the form to display with fields empty, however, the url automatically changes with the previous GET form submission values and the fields on the UI have those values.
Why?

And if I use 'old' in the form, then even when the validation passes or fails, the field on the form is cleared even though I see the value in the url get parameters.

Also, in the notes https://github.com/susanBuck/dwa15-fall2017/blob/master/03_Laravel/21_Validation.md#retaining-form-data-when-validation-fails
`old('title')` throws 'use of undefined constant' error but not with `old('$title')`

**Comment by ALiborio** [comment]
I think this has to do with the validation.
Since it's failing the required field validation, it's redirecting back to what it was previously.
I think if you were able to successfully use the 'old' in the form it would correctly clear the fields, since it would be using the queries that failed validation.

I think one way to work around it is to check if the form is submitted or not. If it's not submitted, skip the validation and just go to the form. There may be a better way, but one way would be to check if `$request->all()` is an empty array, if so just skip the validation and show the form. However, I think using the 'old'  syntax is the way to go.

Can you show us what code you have when you use old('title') and it throws that error?

**Comment by ysrinu** [comment]
yep, this is related to my comment in the post https://github.com/susanBuck/dwa15-fall2017/issues/107
The validation doesn't kickof anywhy if the $request doesn't have the GET query parameters.
When the url is loaded the controller is loaded. When validation fails and redirects (self), it always redirects with query string parameters and hence the validation runs again and fails and loops forever.

'old' could have worked if used from the beginning, but when inserted in between (when there's cache or form already with query string url) then it doesn't help much , from my understanding.
Either way I think there should be a way to fix this without having to be compelled to use 'old' syntax?

**Comment by craterd** [comment]
Why not use 'old' syntax?

**Comment by stbenjam** [comment]
> Now, when I clear the url and only type the base url, I would expect the form to display with fields empty, however, the url automatically changes with the previous GET form submission values and the fields on the UI have those values. Why?

How does the URL automatically change? To include the ? and the values you used before? This almost sounds like your browser autocompleting the rest of the URL from it's history.

**Comment by gabrielsanna** [comment]
@stbenjam Even if the browser saved the URL he had entered and was autocompleting it, that wouldn't fill in the actual fields on his form.

**Comment by ysrinu** [comment]
@stbenjam Good question, and I am sure the browser is not autofilling the url from history. I would know it from other websites/url experience with my browser.
This has to do with route using a cache (probably) $request session for redirect, just a guess

@craterd . 'old' syntax doesn't work either. 'old' is just a convenience if we want to prefer to retain the old values. It is not designed to tackle the redirect. In my case 'old' doesn't work because I added it after the url was already populated with 'bad' GET query string. And this can always happen in real production env. If a dev adds 'old' in code, he cannot ask the users to clear their browser cache/session.
In short, 'old' syntax may be a bandage in certain situation but is not a cure to the problem.

ps: I do love how everybody is putting their thoughts in the forums here, it's nice to share!

**Comment by TheLivingNick** [comment]
Try removing the validation just for a test and see if that fixes the issue. I had a similar problem, and it did turn out the validation was the issue, as it was returning and terminating the code. Removing the validation check (or, later, putting it inside of an if statement that checked to make sure there has been an entry) prevented the issue.

**Comment by stbenjam** [comment]
I think it'd be more helpful to see the code then, you could push it to your repo and share the link with us.

**Comment by susanBuck** [comment]
Closing this post due to inactivity, but feel free to continue the discussion or post a new thread if any problems are still pending.
